### PR TITLE
Allow user to set robot position in simulation on field

### PR DESCRIPTION
**Pull request by loafdog**
User can enter test mode and drag/drop/rotate robot to any place on field.  Or user can update network table entries to set starting pose.  Add the Simulation table to a tab and add all the entries under the table to the tab.  Use the boolean read_start_pose to ignore/use these network table values.  By default they are ignored. If neither Test mode or these network table entries are used, robot uses last known position on field.

### CODE REVIEW

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java [hunk 1]
```diff
@@ -80,7 +84,7 @@ public class DriveTrain extends SubsystemBase {
           Robot.getSysIdConstant("FEED_FORWARD_LINEAR_A").asDouble());
 
   // Defines the field, which is used to display the robot's position on the field in Shuffleboard.
-  private final Field2d field = new Field2d();
+  public final Field2d field = new Field2d();
```
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:87]: Hi Maciej, this was made public, but doesn't seem to be accessed by anyone else.

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java [hunk 2]
```diff
@@ -205,6 +217,42 @@ public DriveTrain() {
     // robot.
     odometry =
         new DifferentialDriveOdometry(navx.getRotation2d(), getLeftDistance(), getRightDistance());
+
+    xSimStart.setNumber(2.0);
+    ySimStart.setNumber(2.0);
+    rotSimStart.setNumber(0.0);
+    readSimStart.setBoolean(false);
```
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:224]: Are these just for debug?

File: src/main/java/bhs/devilbotz/RobotContainer.java [hunk 3]
```diff
@@ -32,7 +32,7 @@ public class RobotContainer {
 
   private final HashMap<AutonomousModes, Command> autoCommands = new HashMap<>();
 
-  private final DriveTrain driveTrain = new DriveTrain();
+  public final DriveTrain driveTrain = new DriveTrain();
```
Comment (BrownGenius) [inline src/main/java/bhs/devilbotz/RobotContainer.java:35]: I see why this was made public.  Is there anyway we can implement this without exposing the driveTrain subsystem?  No biggie if we can't.

**Review by BrownGenius — APPROVED** [review]

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java (hunk 2 above)
Comment (loafdog) [reply src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:224]: fThey are default values to start with.  I just picked them randomly, but we can change in future if we find we are always setting up robot in certain place on the field?

File: src/main/java/bhs/devilbotz/RobotContainer.java (hunk 3 above)
Comment (loafdog) [reply src/main/java/bhs/devilbotz/RobotContainer.java:35]: I'm no OO expert, but I did look at this a bit to see what people do.  `final` means you cannot change it value after initialization. To me a public final variable is same as read only variable.  One way I've seen this done differently is making an accessor method to get the member var needed.  That just seems like lots of extra code that accomplishes same thing.  I like the conciseness using `public final ...`

File: src/main/java/bhs/devilbotz/subsystems/DriveTrain.java (hunk 1 above)
Comment (loafdog) [reply src/main/java/bhs/devilbotz/subsystems/DriveTrain.java:87]: whoops.. probably left over from hacking I did to figure out how to get this working, will put it back way was.

**Review by BrownGenius — APPROVED** [review]

**Review by ParkerMeyers — APPROVED** [review]
Looks good! only thing I see is the use of drivetrain as a public variable and exposing that, but if you think you should do it that way go ahead.

**Pull request merged**